src/poetry/puzzle/provider.py

```diff
@@ -159,6 +160,7 @@ def __init__(
         self._direct_origin_packages: dict[str, Package] = {}
         self._locked: dict[NormalizedName, list[DependencyPackage]] = defaultdict(list)
         self._use_latest: Collection[NormalizedName] = []
+        self._url_authenticator = Authenticator(cache_id="url")
```

Comment:
This assumes that there should be no repository called "url". If we think this could cause potential problems we can also rename it to `_url`